torchgeo/datasets/geo.py

```diff
@@ -1003,7 +1020,10 @@ def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
         samples = []
         for ds in self.datasets:
             if list(ds.index.intersection(tuple(query))):
-                samples.append(ds[query])
+                sample = ds[query]
+                if self.transforms is not None:
+                    sample = self.transforms(sample)
+                samples.append(sample)
```

Comment:
Why does the transform apply to the collated sample in IntersectionDataset but to the uncollated sample in UnionDataset?

Reply:
I had both applied to the collated sample first but mypy gave me a typing error so I changed it to this. I tried again to show you the error and it worked so I was probably doing something wrong in the first place. I'll push the change.